ambry-frontend/src/main/java/com/github/ambry/frontend/AmbrySecurityServiceFactory.java

```diff
@@ -34,20 +35,23 @@
   private final FrontendMetrics frontendMetrics;
   private final UrlSigningService urlSigningService;
   private final StorageQuotaService storageQuotaService;
+  private final QuotaManager quotaManager;
 
   public AmbrySecurityServiceFactory(VerifiableProperties verifiableProperties, ClusterMap clusterMap,
       AccountService accountService, UrlSigningService urlSigningService, IdSigningService idSigningService,
-      AccountAndContainerInjector accountAndContainerInjector, StorageQuotaService storageQuotaService) {
+      AccountAndContainerInjector accountAndContainerInjector, StorageQuotaService storageQuotaService,
+      QuotaManager quotaManager) {
```

Comment:
Does StorageQuotaService not have its own reference to the QuotaManager?  It seems odd that they both need to be supplied.

Reply:
Yes in the complete implementation, StorageQuota will be part of the QuotaManager. But as noted in the PR comment, I haven't merged StorageQuota with the QuotaManager yet, as I don't want to interfere with Justin's testing, until this thing is ready and is working end to end in closed source.
I will do that in a followup PR.